packages/sourcecred/src/core/ledger/computeDistribution.js

```diff
@@ -40,20 +40,21 @@ export function computeDistribution(
 }
 
 export function _allocationIdentities(
-  accountsData: CredAccountData,
+  credGrainData: CredGrainView,
   effectiveTimestamp: TimestampMs
 ): $ReadOnlyArray<AllocationIdentity> {
-  const activeAccounts = accountsData.accounts.filter(
-    ({account}) => account.active
-  );
-  const allocationIdentities = activeAccounts.map((x) => ({
-    id: x.account.identity.id,
-    paid: x.account.paid,
-    cred: x.cred,
+  const activeParticipants = credGrainData
+    .participants()
+    .filter((participant) => participant.active);
+
+  const allocationIdentities = activeParticipants.map((x) => ({
+    id: x.identity.id,
+    paid: x.grainEarned,
+    cred: x.credPerInterval,
   }));
-  const numIntervals = accountsData.intervals.filter(
-    (x) => x.endTimeMs <= effectiveTimestamp
-  ).length;
+  const numIntervals = credGrainData
+    .intervals()
+    .filter((x) => x.endTimeMs <= effectiveTimestamp).length;
   const timeSlicedAllocationIdentities = allocationIdentities.map((x) => ({
```

Comment:
A nicer way to do this is to use the `withTimeScope` method on the CredGrainView at the top of this block.

Reply:
fixed

Reply:
So actually, I was thinking something like this:
```
const activeParticipants = credGrainData
    .withTimeScope(0, effectiveTimestamp)
    .participants()
    .filter((participant) => participant.active);

const allocationIdentities = activeParticipants.map((x) => ({
    id: x.identity.id,
    paid: x.grainEarned,
    cred: x.credPerInterval,
  }));

return allocationIdentities;
```
as this removes the need for 56-64

Reply:
Fixed